### PR TITLE
Add 'Key Definitions' section to the UG

### PR DESCRIPTION
**Pull request by nur-haziq**
Added a **'Key Definitions** section to the UG, which consists of the following subsections:
- Item
- Flags
- Placeholders (viewing in source diff is cancer, you may view in rich diff)
- Command format

This is to aid a non-technical user in understanding what these terms are, before jumping into the `Features` section of the UG, where they are met with repeated use of commands.

### CODE REVIEW

File: docs/images/UG_CommandFormat.png
Comment (YHWong20) [inline docs/images/UG_CommandFormat.png:1]: Great addition to the UG!

File: docs/UserGuide.md [hunk 1]
```diff
@@ -148,46 +156,121 @@ types of items in your inventory list. They are:
...
+> Please refer to the subsequent [Command Format](#command-format) section to see how Flags and Placeholders are used 
+> together.
+
+| Flag  | Corresponding Placeholder           |
+|-------|-------------------------------------|
+| `-n`  | ITEM_NAME                           |
+| `-i`  | ITEM_INDEX                          |
+| `-d`  | ITEM_DESCRIPTION                    |
+| `-q`  | QUANTITY                            |
+| `-e`  | EXPIRY_DATE (Perishable items only) |
```
Comment (YHWong20) [inline docs/UserGuide.md:202]: Shall we bolden the `(Perishable items only)` part for emphasis?
```suggestion
| `-e`  | EXPIRY_DATE **(Perishable items only)** |
```

File: docs/UserGuide.md [hunk 2]
```diff
@@ -148,46 +156,121 @@ types of items in your inventory list. They are:
...
+| `-c`  | COST_PRICE                          |
+| `-s`  | SALE_PRICE (Retail items only)      |
```
Comment (YHWong20) [inline docs/UserGuide.md:204]: Similar suggestion here, as above.
```suggestion
| `-s`  | SALE_PRICE **(Retail items only)**      |
```

File: docs/UserGuide.md [hunk 3]
```diff
@@ -148,46 +156,121 @@ types of items in your inventory list. They are:
...
+| `ITEM_DESCRIPTION` | `-d`               | ITEM_DESCRIPTION provides details about the item.<br>**Info:**<ul><li>Can include most characters and symbols.</li><li>Should give a clear idea of the item's features or use.</li></ul>**Valid Examples:** `Black and white running sneakers`, `High-yield color cartridge`<br>**Invalid Examples:** (Any unclear or overly lengthy description)                               |
+| `ITEM_QUANTITY`    | `-q`               | ITEM_QUANTITY is how much of the item is in stock.<br>**Info:**<ul><li>Must be a non-negative integer.</li></ul>**Valid Examples:** `350`, `20`<br>**Invalid Examples:** `-10`, `100.5`                                                                                                                                                                                         |
```
Comment (YHWong20) [inline docs/UserGuide.md:229]: `integer` is a somewhat technical term. We can consider calling it a `whole number` instead.

**Review by YHWong20 — APPROVED** [review]
Looks great. Thanks for making the command format section more detailed!

File: docs/UserGuide.md [hunk 4]
```diff
@@ -148,46 +156,121 @@ types of items in your inventory list. They are:
...
+| `INDEX`            | `-i`               | The INDEX is the number to the left of the item name when the `list` command is called.<br>**Info:**<ul><li>INDEX must be a whole number larger than 0.</li><li>It is dependent on the current view of the item list.</li><li>Do not use thousands separators like `,` in INDEX.</li></ul>**Valid Examples:** `1`, `124`<br>**Invalid Examples:** `-1`, `1.5`, `1,132`, `1 132` |
+| `THRESHOLD`        | `-t`               | THRESHOLD is the minimum stock before an alert is issued.<br>**Info:**<ul><li>Must be a non-negative integer.</li><li>If not specified, `1` is used by default.</li></ul>**Valid Examples:** `50`, `10`<br>**Invalid Examples:** `-50`, `10.5`                                                                                                                                  |
```
Comment (YHWong20) [inline docs/UserGuide.md:234]: Same here, maybe we could replace `integer` with `whole number`.

File: docs/UserGuide.md (hunk 1 above)
Comment (nur-haziq) [reply docs/UserGuide.md:202]: Good idea, will do that!

File: docs/UserGuide.md (hunk 3 above)
Comment (nur-haziq) [reply docs/UserGuide.md:229]: Changed it, thanks!

**Pull request merged**